frontend/pulse3d/src/pages/users-info.js

```diff
@@ -0,0 +1,257 @@
+import DashboardLayout from "@/components/layouts/DashboardLayout";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Paper from "@mui/material/Paper";
+import { useState, useEffect } from "react";
+import Row from "@/components/admin/UsersRow";
+import styled from "styled-components";
+import ModalWidget from "@/components/basicWidgets/ModalWidget";
+
+const PageContainer = styled.div`
+  width: 80%;
+`;
+
+const Pagination = styled.div`
+  width: 100%;
+  background-color: var(--dark-gray);
+  display: flex;
+  justify-content: right;
+  padding: 1rem 1rem;
+`;
+
+const Container = styled.div`
+  display: flex;
+  position: relative;
+  justify-content: start;
+  padding: 0% 3% 3% 3%;
+  flex-direction: column;
+  margin-top: 1rem;
+`;
+
+export default function UserInfo() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [rows, setRows] = useState([]);
+  const [currentRows, setCurrentRows] = useState([]);
+  const [users, setUsers] = useState([]);
+  const [actionAlertVisible, setActionAlertVisible] = useState(false);
+  const [modalMessage, setModalMessage] = useState();
+  const [actionToPreform, setActionToPreform] = useState("");
+  const [confirm, setConfirm] = useState(false);
+  const [emailToDelete, setEmailToDelete] = useState("");
+
+  const deleteUser = async () => {
+    fetch(`https://curibio.com/user-actions/${emailToDelete}`, {
+      method: "DELETE",
+    });
+  };
+  const deactivateUser = async () => {
+    fetch(`https://curibio.com/user-actions/${emailToDelete}`, {
+      method: "PUT",
+      body: JSON.stringify({ action_type: "deactivate" }),
+    });
+  };
+
+  const userActionSelection = async (option, name, email) => {
+    if (option === 1) {
+      setModalMessage(`Are you sure you would like to deactivate ${name}?`);
+      setActionToPreform("deactivate");
+      setEmailToDelete(email);
+    } else if (option === 0) {
+      setModalMessage(`Are you sure you would like to delete ${name}?`);
+      setActionToPreform("delete");
+      setEmailToDelete(email);
+    }
+  };
+
+  const getAllUsers = async () => {
+    try {
+      const response = await fetch("https://curibio.com/users");
+      if (response && response.status === 200) {
+        const usersJson = await response.json();
+        const userList = [];
+        for (const user of usersJson) {
+          userList.push(user);
+        }
```

Comment:
I don't think you need to create a new array and push the user in.. the usersJson looks to be an array already?